src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -391,21 +392,26 @@ abstract class AbstractCheckTask(
                 messageContainer.metadataMessage,
                 it.message, it.comparatorSettings,
                 matchNames
-            ).also { comparisonResult ->
+            )?.also { comparisonResult ->
                 LOGGER.debug("Metadata comparison result\n {}", comparisonResult)
             }
         }
         if (metadataFilter != null && metadataComparisonResult == null) {
             if (LOGGER.isDebugEnabled) {
                 LOGGER.debug("Metadata for message {} does not match the filter by key fields. Skip message checking",
-                    shortDebugString(messageContainer.protoMessage.metadata.id))
+                        messageContainer.protoMessage.metadata.id.toJson())
             }
             return AggregatedFilterResult.EMPTY
         }
         val comparisonResult: ComparisonResult? = messageFilter.let {
             MessageComparator.compare(messageContainer.sailfishMessage, it.message, it.comparatorSettings, matchNames)
         }
-        LOGGER.debug("Compare message '{}' result\n{}", messageContainer.sailfishMessage.name, comparisonResult)
+
+        if (comparisonResult == null) {
+            LOGGER.debug("Comparison result for message '{}' does not match the filter by key fields or message type", messageContainer.sailfishMessage.name)
```

Comment:
I would also add the expected message type from the filter. It will show if the result is null because types are different